include/deal.II/base/mpi.h

```diff
@@ -1745,7 +1745,7 @@ namespace Utilities
      * not satisfied.
      */
     template <typename T>
-    const MPI_Datatype mpi_type_id_for_type =
+    static const MPI_Datatype mpi_type_id_for_type =
       internal::MPIDataTypes::mpi_type_id(
         static_cast<std::remove_cv_t<std::remove_reference_t<T>> *>(nullptr));
```

Comment:
I prefer this variant over marking the variable `extern` because it allows the compiler to potentially optimize the data structure/variable away and replace it with a constant.

Reply:
We could also switch to shiny new C++17 `inline` here. (On a sidenote: I would really love to actually use `constexpr` here - but that's only possible with how mpich implemented `MPI_Datatype` and not how OpenMPI implemented the type and the macros).

Reply:
I'd prefer `inline const`.

Reply:
Yes, you can't do `constexpr` because what you have in openmpi is not known at compile time.

I still think `inline` is the right way to go.

Reply:
converting to `inline const`.